File: news-api/src/main/java/com/news/search/infrastructure/NaverWebRepositoryImpl.java
package com.news.search.infrastructure;

import com.news.feign.NaverClient;
import com.news.model.Item;
import com.news.model.NaverWebResponse;
import com.news.search.service.port.WebRepository;
import com.news.search.service.response.PageQueryResult;
import com.news.search.service.response.SearchQueryResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.List;

@RequiredArgsConstructor
@Repository
public class NaverWebRepositoryImpl implements WebRepository {
    private final NaverClient naverClient;

    @Override
    public PageQueryResult<SearchQueryResponse> search(String query, int page, int size) {
        NaverWebResponse response = naverClient.search(query, page, size);
        List<SearchQueryResponse> responses = response.items().stream()
                .map(this::convertToSearchResponse)
                .toList();
        return new PageQueryResult<>(page, size, response.total(), responses);
    }

    private SearchQueryResponse convertToSearchResponse(Item item){
        return SearchQueryResponse.builder()
                .title(item.title())
                .link(item.link())
                .description(item.description())
                .build();
    }
}

File: news-api/src/main/kotlin/com/search/websearch/infrastructure/NaverWebRepositoryImpl.kt
package com.search.websearch.infrastructure

import com.search.feign.NaverClient
import com.search.model.Item
import com.search.websearch.service.port.WebRepository
import com.search.websearch.service.result.WebSearchPageResult
import com.search.websearch.service.result.WebSearchResult
import org.springframework.stereotype.Repository

@Repository
class NaverWebRepositoryImpl(
        val naverClient: NaverClient
) : WebRepository{
    override fun search(query: String, page: Int, size: Int): WebSearchPageResult<WebSearchResult> {
        val searchResult = naverClient.search(query, page, size)
        val contents = searchResult.items.map { toWebSearchResult(it) }.toList()
        return WebSearchPageResult(page, size, searchResult.total, contents)
    }

    private fun toWebSearchResult(item: Item): WebSearchResult{
        return WebSearchResult(
                title = item.title,
                link = item.link,
                description = item.description
        )
    }
}
